File: solutions/1231-replace-elements-with-greatest-element-on-right-side/solution.py
class Solution(object):
    def replaceElements(self, arr):
        """
        :type arr: List[int]
        :rtype: List[int]
        """
        for i in range(len(arr)-1,0,-1):
            if arr[i]>arr[i-1]:
                arr[i-1]=arr[i]
        del arr[0]
        arr.append(-1)
        return arr

File: solutions/1231-replace-elements-with-greatest-element-on-right-side/solution.java
class Solution {
    public int[] replaceElements(int[] arr) {
        for(int i=arr.length-1;i>0;i--)
            if(arr[i]>arr[i-1])
                arr[i-1]=arr[i];
        for(int i=1;i<arr.length;i++)
            arr[i-1]=arr[i];
        arr[arr.length-1]=-1;
        return arr;
    }
}
